crates/core/src/record/mod.rs

```diff
@@ -1,8 +1,19 @@
 use std::sync::{Arc, RwLock};
 
-use hashbrown::{HashMap, HashSet};
+use hashbrown::HashMap;
 
-use crate::{module::{ModuleId, ModuleType}, plugin::{PluginAnalyzeDepsHookResultEntry, ResolveKind}};
+use crate::{
+  module::{ModuleId, ModuleType},
+  plugin::PluginAnalyzeDepsHookResultEntry,
+};
+
+#[derive(Debug, Clone)]
+pub enum Stage {
+  Init,
```

Comment:
what does `Init` mean

Reply:
but we only have 2 stages: build and generate, I think we can rename it to avoid name conflict

Reply:
No use stage to name, use something like status, mode, phase ？Currently only used to identify whether it is an HMR, but it may be expanded to a stage in the future

Reply:
rename filed stage as trigger